#request constants
float64 posX
float64 posY
------------
#response constants
float64 S
